File: examples/python/dynamodb_update_item.py
"""Actualiza el atributo "estado" de un item existente sin sobrescribir el resto.

Uso: python3 dynamodb_update_item.py <tabla> <id> <nuevo-estado>
"""
import json
import sys

import boto3

dynamodb = boto3.client(
    "dynamodb",
    endpoint_url="http://localhost:4566",
    aws_access_key_id="test",
    aws_secret_access_key="test",
    region_name="us-east-1",
)


def main():
    if len(sys.argv) < 4:
        raise SystemExit("Uso: python3 dynamodb_update_item.py <tabla> <id> <nuevo-estado>")
    table_name, item_id, nuevo_estado = sys.argv[1], sys.argv[2], sys.argv[3]

    result = dynamodb.update_item(
        TableName=table_name,
        Key={"id": {"S": item_id}},
        UpdateExpression="SET estado = :nuevoEstado",
        ExpressionAttributeValues={":nuevoEstado": {"S": nuevo_estado}},
        ReturnValues="ALL_NEW",
    )
    print("Item actualizado:")
    print(json.dumps(result["Attributes"], indent=2, ensure_ascii=False))


if __name__ == "__main__":
    main()

File: examples/rust/dynamodb_update_item.rs
// Actualiza el atributo "estado" de un item existente sin sobrescribir el resto.
// Uso: cargo run --bin dynamodb_update_item -- <tabla> <id> <nuevo-estado>

use aws_config::BehaviorVersion;
use aws_credential_types::Credentials;
use aws_sdk_dynamodb::{
    config::Region,
    types::{AttributeValue, ReturnValue},
    Client, Config,
};

#[tokio::main]
async fn main() -> Result<(), aws_sdk_dynamodb::Error> {
    let args: Vec<String> = std::env::args().collect();
    if args.len() < 4 {
        eprintln!("Uso: cargo run --bin dynamodb_update_item -- <tabla> <id> <nuevo-estado>");
        std::process::exit(1);
    }
    let table_name = &args[1];
    let id = &args[2];
    let nuevo_estado = &args[3];

    let config = Config::builder()
        .behavior_version(BehaviorVersion::latest())
        .endpoint_url("http://localhost:4566")
        .region(Region::new("us-east-1"))
        .credentials_provider(Credentials::new("test", "test", None, None, "floci"))
        .build();
    let client = Client::from_conf(config);

    let result = client
        .update_item()
        .table_name(table_name)
        .key("id", AttributeValue::S(id.clone()))
        .update_expression("SET estado = :nuevoEstado")
        .expression_attribute_values(":nuevoEstado", AttributeValue::S(nuevo_estado.clone()))
        .return_values(ReturnValue::AllNew)
        .send()
        .await?;

    println!("Item actualizado:");
    println!("{:#?}", result.attributes());
    Ok(())
}
